src/main/java/com/khalsa_ji/ems/repository/EmployeeRepository.java

```diff
@@ -0,0 +1,89 @@
+//  Waheguru Ji!
+
+package com.khalsa_ji.ems.repository;
+
+import com.khalsa_ji.ems.Employee;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * The interface {@code EmployeeRepository} defines various methods to play around with the
+ * attributes and tuples of the <em>Employee</em> table described by the {@code Employee} class.
+ *
+ * @author Ravikiran Singh
+ * @version 1.0
+ * @since 1.0
+ * @see Employee
+ */
+
+public interface EmployeeRepository extends JpaRepository<Employee, Long> {
+    /**
+     * Method to fetch an instance of {@code Employee} class specified by its id
+     *
+     * @param ID Employee's id
+     * @return An instance of {@code Employee} class of specified id
+     * @see Employee
+     */
+
+    Employee findByEmployeeID(Long ID);
+
+    /**
+     * Method to fetch an instance of {@code Employee} class specified by its manager id
+     *
+     * @param managerID Employee's manager id(Employee id of the manager)
+     * @return An instance of {@code Employee} class of specified manager id
+     * @see Employee
+     */
+
+    Employee findByManagerID(Long managerID);
+
+    /**
+     * Method to fetch a list of {@code Employee} class instances of specified manager id
+     *
+     * @param managerID Employee's manager id(Employee id of the manager)
+     * @return List of {@code Employee} class instances of specified manager id
+     * @see Employee
+     */
+
+    List<Employee> findAllByManagerID(Long managerID);
+
+    /**
+     * Method to fetch all instances of {@code Employee} class and order them by their designation's level id and employee name
+     *
+     * @return List of {@code Employee} class instances in an ordered manner
+     * @see Employee
+     */
+
+    List<Employee> findAllByOrderByJobID_levelIDAscEmployeeNameAsc();
```

Comment:
Agreed, but can't help with that, as it is one of the Spring Data JPA _Query Methods_ convention, which gets automatically implemented.
This query method got too long because of its complex nature regarding sorting the list of all employees found by their designation's level id and if that is found to be same then sorting based on employee's name.